probeye/definition/inference_problem.py

```diff
@@ -1303,3 +1303,24 @@ def check_problem_consistency(self):
                     f"The globally defined experiment '{exp_name}' does not appear in "
                     f"any of the likelihood models!"
                 )
+    
+    def check_parameters_domain(self, theta):
+        """
+        Checks weather the given values of the latent parameters are within the
+        specified domain.
+
+        Parameters
+        ----------
+        theta
+            A numeric vector or tensor, which contains the current values of all latent parameters.
+
+        Returns
+        -------
+        True if all values are within the specified domain. Otherwise, False is returned.
+        """
+        for theta_idx, theta_name in enumerate(self.get_theta_names()):
+            if theta[theta_idx] < self._parameters[theta_name].domain[0][0]:
+                return False
+            elif theta[theta_idx] > self._parameters[theta_name].domain[0][1]:
+                return False
+        return True       
```

Comment:
Your code works perfectly for scalar parameters! You probably didn't know, but you can also define vector-parameters with 2 or more components. I'm suggesting the following modification to make it work with them too:
```suggestion
        for theta_name in self.get_theta_names():
            theta_dim = self._parameters[theta_name].dim
            theta_idx = self._parameters[theta_name].index
            for i in range(theta_dim):
                if theta[theta_idx + i] < self._parameters[theta_name].domain[i][0]:
                    return False
                elif theta[theta_idx + i] > self._parameters[theta_name].domain[i][1]:
                    return False
        return True
        
```